### PR TITLE
chore: Added rework of filters and split them out into separate files

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR introduces a refactoring of the filters for better maintainability. It introduces 5 utility files which are somewhat copy/paste with a little rewrite to ES6:

- `bindings` - contains all the binding support functions used for NATS.
- `channelname` - contains the functions for realizing the channel name based on provided parameters.
- `general` -  contains the general support functions for typescript
- `index` - wrapper which exports all the functions
- `parameters` - contains the functions for realizing the parameters to a function in the template.

### CODE REVIEW

File: __react/utils/bindings.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+export function includeUnsubAfterForSubscription(message) {
```
Comment (fmvilas) [inline __react/utils/bindings.js:1]: It would be great to have some docs explaining what this function is doing.

File: __react/utils/bindings.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    return `subscribeOptions.queue = '${message.bindings().nats().queue()}';`;
+  }
+  return '';
+}
+
+/**
+ * Does an object have bindings
+ */
+export function hasNatsBindings(obj) {
+  return obj.bindings && obj.bindings.nats;
```
Comment (fmvilas) [inline __react/utils/bindings.js:18]: Shouldn't it be like this?

```suggestion
  return obj.bindings() && obj.bindings().nats();
```

File: __react/utils/bindings.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+}
+  
+/**
+   * is the channel a publish and subscribe type if nothing is specified default to being pubsub type 
+   */
```
Comment (fmvilas) [inline __react/utils/bindings.js:23]: Two things:

1. Indentation.
2. I don't understand this explanation 😄 I have the feeling there's some punctuation mark missing.

File: __react/utils/bindings.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+export function isPubsub(channel) {
+  const tempChannel = channel._json;
+  if (
+    !tempChannel.bindings || 
+      !tempChannel.bindings.nats ||
+      !tempChannel.bindings.nats.is || 
+      tempChannel.bindings.nats.is === 'pubsub') {
```
Comment (fmvilas) [inline __react/utils/bindings.js:30]: Same as above. Shouldn't it access `bindings().nats()` instead of `bindings.nats`?

**Review by fmvilas — CHANGES_REQUESTED** [review]

File: __react/utils/bindings.js (hunk 2 above)
Comment (jonaslagoni) [reply __react/utils/bindings.js:18]: Yea you are right, I should address this 😄

File: __react/utils/bindings.js (hunk 1 above)
Comment (jonaslagoni) [reply __react/utils/bindings.js:1]: You are right, I'll add jsdocs to all the functions.

File: __react/utils/bindings.js (hunk 4 above)
Comment (jonaslagoni) [reply __react/utils/bindings.js:30]: Should actually use the other function as well 😄

File: __react/utils/bindings.js (hunk 2 above)
Comment (fmvilas) [reply __react/utils/bindings.js]: 👍 There are others too.

File: __react/utils/bindings.js (hunk 3 above)
Comment (jonaslagoni) [reply __react/utils/bindings.js]: I have added a new definition :)

File: __react/utils/bindings.js (hunk 2 above)
Comment (magicmatatjahu) [reply __react/utils/bindings.js]: We don't have something like `nats()` :) https://github.com/asyncapi/parser-js/blob/master/lib/mixins/bindings.js

File: __react/utils/bindings.js (hunk 4 above)
Comment (magicmatatjahu) [reply __react/utils/bindings.js]: We don't have something like `nats()` :) https://github.com/asyncapi/parser-js/blob/master/lib/mixins/bindings.js

File: __react/utils/bindings.js (hunk 2 above)
Comment (jonaslagoni) [reply __react/utils/bindings.js]: Nope, I used `binding('nats')` instead :)

File: __react/utils/bindings.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
+
+
+/**
+ * Wrapper to include subscriptions option code if specified in the spec.
+ * 
+ * @param {*} message to check for queue bindings on 
+ */
+export function includeUnsubAfterForSubscription(message) {
+  if (message.hasBinding('nats') && message.bindings().nats.unsubAfter) {
+    return `subscribeOptions.max = '${message.bindings().nats.unsubAfter}';`;
```
Comment (fmvilas) [inline __react/utils/bindings.js:10]:
```suggestion
    return `subscribeOptions.max = '${message.binding('nats').unsubAfter}';`;
```

**Review by fmvilas — CHANGES_REQUESTED** [review]

File: __react/utils/bindings.js (hunk 5 above)
Comment (jonaslagoni) [reply __react/utils/bindings.js]: 👼

**Review by fmvilas — APPROVED** [review]
:shipit:

**Pull request merged**